lib/src/component_declaration/component_base.dart

```diff
@@ -124,6 +124,16 @@ abstract class UiComponent<TProps extends UiProps> extends react.Component {
     );
   }
 
+  /// Validation method called during [componentWillReceiveProps], and [componentWillMount].
+  ///
+  /// Use this as an opportunity validate props during the correct lifecycle of the component.
```

Comment:
#nit I think this comment could be a little more descriptive.
```dart
/// Throws a [PropError] if [appliedProps] are invalid.
/// 
/// This is called automatically with during [componentWillReceiveProps] and [componentWillMount],
/// and can also be called manually for custom validation.
/// 
/// Override with a custom implementation to easily add validation (and don't forget to call super!)
///
///     @mustCallSuper
///     void validateProps(Map appliedProps) {
///       super.validateProps(appliedProps);
///       
///       var tProps = typedPropsFactory(appliedProps);
///       if (tProps.items.length.isOdd) {
///         throw new PropError.value(tProps.items, 'items', 'must have an even number of items, because reasons');
///       }
///     }
```